storage/src/robocorp/storage/_client.py

```diff
@@ -68,9 +68,22 @@ def _handle_error(response: Response):
 
         return _handle_error
 
-    def list_assets(self) -> List[Asset]:
+    def list_assets(
+        self, *, limit: Optional[int], next_page: Optional[str]
+    ) -> AssetsResponse:
         """Retrieve list of assets."""
-        return self._client.get("").json()
+        if limit and next_page:
+            raise ValueError(
+                "Choose between a limit (for the initial request) or the next page URL"
+                " (for any subsequent request)"
+            )
+        params = None
+        url = ""
+        if limit:
+            params = {"limit": limit}
+        elif next_page:
+            url = next_page
+        return self._client.get(url, params=params).json()
```

Comment:
If there's no `url` nor `limit` passed, a default API limit is implicitly used when requesting all the available assets for the first time.

Otherwise we start with a `limit` and as long as we get a `next_page` we'll continue to retrieve assets from that one.